### ISSUE TITLE
Keep users operations in blockchain

### ISSUE THREAD

**Issue by radumarias** [xoriors/rencfs#243]
We can save user operations, hashes of files, users with whom he shares files, dates, IPs, etc.

The goal is to save all operations like a log file but keep that in the blockchain. We will save just a hash of the filename, time, and operation performed. Maybe we can use some public blockchain?

Alternatively, we can save it in [Mainline DHT](https://en.wikipedia.org/wiki/Mainline_DHT), maybe by encrypting the data.

# Structure notes

- Create a dedicated crate for this that uses rencfs (the core) as a lib. You can create attractions and common minimalistic generic functionality in the core but keep the core more generic

**Comment by longregen** [comment]
Hi there!

I'm a big enthusiast of blockchain-based tech, yet I don't fully understand the goal of this issue. Maybe you meant an append-only log with some sort of hashing tree structure?

**Comment by radumarias** [comment]
Yes, the goal is to save all operations like a log file but keep that in blockchain. We will save just hash of filename, time and operation performed. Maybe we can use some pub blockchain?

Alternatively we can save in [Mainline DHT](https://en.wikipedia.org/wiki/Mainline_DHT) maybe encrypting the data.

**Comment by longregen** [comment]
Isn't this a little bit of a privacy nightmare?

**Comment by radumarias** [comment]
in what way? if we encrypt the data or keep it in our network we should be fine

**Comment by radumarias** [comment]
> Isn't this a little bit of a privacy nightmare?

@longregen can you elaborate?
